### PR TITLE
feature: install submariner broker and globalnet config on hub cluster

### PR DESCRIPTION
**Pull request by SataQiu**
Signed-off-by: SataQiu <shidaqiu2018@gmail.com>

/cc @sridhargaddam

### CODE REVIEW

**Comment by SataQiu** [comment]
Hi @sridhargaddam 

Could you please take a look?

**Comment by sridhargaddam** [comment]
> Hi @sridhargaddam
> 
> Could you please take a look?

Sure @SataQiu. I'll take a look at the PR tomorrow. Thanks.

**Comment by sridhargaddam** [comment]
> Hi @sridhargaddam
> 
> Could you please take a look?

Hello @SataQiu, May I know how you are testing this feature? Are you using KIND or ACM GUI? 
If it's KIND, what options did you pass?

File: pkg/hub/submarinerbroker/controller.go [hunk 1]
```diff
@@ -115,9 +131,57 @@ func (c *submarinerBrokerController) sync(ctx context.Context, syncCtx factory.S
...
 }
 
+func buildBrokerConfig(clusterSet *clusterv1beta1.ManagedClusterSet) (*brokerConfig, error) {
+	config := &brokerConfig{
+		SubmarinerNamespace:         brokerinfo.GenerateBrokerName(clusterSet.Name),
+		Components:                  []string{"service-discovery", "connectivity"},
+		DefaultCustomDomains:        nil,
+		GlobalnetCIDRRange:          "242.0.0.0/8",
+		DefaultGlobalnetClusterSize: 65536,
+		GlobalnetEnabled:            false,
```
Comment (sridhargaddam) [inline pkg/hub/submarinerbroker/controller.go:141]: I manually modified this to "true" and wanted to verify if globalnet is getting deployed with KIND Clusters (i.e., `make images`, `make clusters` and `make demo`). However, globalnet is not deployed.

Also, broker-globalnet-cm as well as broker objects are not created on the broker cluster.  

Can you please let us know how you are testing this change?

**Review by sridhargaddam — CHANGES_REQUESTED** [review]

File: pkg/hub/submarinerbroker/controller.go [hunk 2]
```diff
@@ -115,9 +131,57 @@ func (c *submarinerBrokerController) sync(ctx context.Context, syncCtx factory.S
...
+	}
+
+	if v, ok := clusterSet.Annotations[brokerComponentsAnnotationKey]; ok {
```
Comment (sridhargaddam) [inline pkg/hub/submarinerbroker/controller.go:144]: @skitt @tpantelis do you have any feedback on this approach of using annotations on the ManagedClusterSet object to enable certain features?

Comment (tpantelis) [reply pkg/hub/submarinerbroker/controller.go:144]: I don't know if this is a recommended approach and if other apps do it. For the submariner install on the managed clusters, we have a `SubmarinerConfig` CRD to communicate the user config so perhaps we should create an equivalent `BrokerConfig` CRD to be consistent.

Comment (SataQiu) [reply pkg/hub/submarinerbroker/controller.go:144]: I'm hesitant about this place, too. 
If `BrokerConfig` is a more acceptable solution, I can try it. @tpantelis @sridhargaddam

Comment (tpantelis) [reply pkg/hub/submarinerbroker/controller.go:144]: yeah let's try that. We don't own the `ManagedClusterSet` so our annotations could cause extraneous updates for others and possibly get overwritten.

Comment (Jaanki) [reply pkg/hub/submarinerbroker/controller.go:144]: Ideal way to create `Broker` on hub cluster is to install `submariner-operator` on the hub cluster and let the operator handle `broker` creation. `submariner` is created in the spoke in the similar fashion.

Comment (tpantelis) [reply pkg/hub/submarinerbroker/controller.go:144]: yeah it seems like we should be able to use the [Broker](https://github.com/submariner-io/submariner-operator/blob/devel/api/submariner/v1alpha1/submariner_types.go#L191) CRD and install the `submariner-operator`.

**Pull request closed without merging**